src/components/modebar/buttons_config.js

```diff
@@ -0,0 +1,257 @@
+        category: '2d',
+        group: 'drag',
+        title: 'Pan',
+        attr: 'dragmode',
+        val: 'pan',
+        icon: 'pan',
+        click: 'handleCartesian'
+    },
+    zoomIn2d: {
+        category: '2d',
+        group: 'zoom',
+        title: 'Zoom in',
+        attr: 'zoom',
+        val: 'in',
+        icon: 'zoom_plus',
+        click: 'handleCartesian'
+    },
+    zoomOut2d: {
+        category: '2d',
+        group: 'zoom',
+        title: 'Zoom out',
+        attr: 'zoom',
+        val: 'out',
+        icon: 'zoom_minus',
+        click: 'handleCartesian'
+    },
+    autoScale2d: {
+        category: '2d',
+        group: 'zoom',
+        title: 'Autoscale',
+        attr: 'zoom',
+        val: 'auto',
+        icon: 'autoscale',
+        click: 'handleCartesian'
+    },
+    resetScale2d: {
+        category: '2d',
+        group: 'zoom',
+        title: 'Reset axes',
+        attr: 'zoom',
+        val: 'reset',
+        icon: 'home',
+        click: 'handleCartesian'
+    },
+
+    // cartesian only
+    hoverClosest2d: {
+        category: 'cartesian',
+        group: 'hover',
+        title: 'Show closest data on hover',
+        attr: 'hovermode',
+        val: 'closest',
+        icon: 'tooltip_basic',
+        gravity: 'ne',
+        click: 'handleCartesian'
+    },
+    hoverCompare2d: {
+        category: 'cartesian',
+        group: 'hover',
+        title: 'Compare data on hover',
+        attr: 'hovermode',
+        val: function(graphInfo) {
+            return graphInfo._fullLayout._isHoriz ? 'y' : 'x';
+        },
+        icon: 'tooltip_compare',
+        gravity: 'ne',
+        click: 'handleCartesian'
+    },
+
+    // gl3d
+    zoom3d: {
+        category: 'gl3d',
+        group: 'drag',
+        title: 'Zoom',
+        attr: 'dragmode',
+        val: 'zoom',
+        icon: 'zoombox',
+        click: 'handleDrag3d'
+    },
+    pan3d: {
+        category: 'gl3d',
+        group: 'drag',
+        title: 'Pan',
+        attr: 'dragmode',
+        val: 'pan',
+        icon: 'pan',
+        click: 'handleDrag3d'
+    },
+    orbitRotation: {
+        category: 'gl3d',
+        group: 'drag',
+        title: 'orbital rotation',
+        attr: 'dragmode',
+        val: 'orbit',
+        icon: '3d_rotate',
+        click: 'handleDrag3d'
+    },
+    tableRotation: {
+        category: 'gl3d',
+        group: 'drag',
+        title: 'turntable rotation',
+        attr: 'dragmode',
+        val: 'turntable',
+        icon: 'z-axis',
+        click: 'handleDrag3d'
+    },
+    resetCameraDefault3d: {
+        category: 'gl3d',
+        group: 'zoom',
+        title: 'Reset camera to default',
+        attr: 'resetDefault',
+        icon: 'home',
+        click: 'handleCamera3d'
+    },
+    resetCameraLastSave3d: {
+        category: 'gl3d',
+        group: 'zoom',
+        title: 'Reset camera to last save',
+        attr: 'resetLastSave',
+        icon: 'movie',
+        click: 'handleCamera3d'
```

Comment:
@alexcjohnson @cldougl 

This would be a good time to *standardize* the modebar buttons nomenclature. 

Any suggestions are welcomed.